app_dart/test/service/branch_service_test.dart

```diff
@@ -53,6 +54,22 @@ void main() {
       when(githubService.github.repositories).thenReturn(mockRepositoriesService);
     });
 
+    test('return 404 when branch version file does not exist', () async {
+      final gh.Branch candidateBranch = generateBranch(3, name: 'flutter-3.4-candidate.5', sha: '789dev');
+      when(mockRepositoriesService.listBranches(any)).thenAnswer((Invocation invocation) {
+        return Stream.fromIterable([candidateBranch]);
+      });
+      when(mockRepositoriesService.getContents(any, any)).thenAnswer((Invocation invocation) {
```

Comment:
`thenAnswer` is for futures, in this case you can use `thenReturn`

```suggestion
      when(mockRepositoriesService.getContents(any, any)).thenReturn((_) =>
```

Reply:
umm Couldn't get `thenReturn` to work, ended up using `thenThrow` for the Error type